ionc/ion_int.c

```diff
@@ -1599,28 +1611,63 @@ SIZE _ion_int_abs_bytes_signed_length_helper(const ION_INT *iint)
     return _ion_int_abs_bytes_length_helper_helper(iint, /*is_signed=*/TRUE);
 }
 
-
 iERR _ion_int_to_int64_helper(ION_INT *iint, int64_t *p_int64)
 {
     iENTER;
     II_DIGIT *digits, *end, digit;
-    int64_t   value = 0;
+    uint64_t magnitude = 0;
 
     digits = iint->_digits;
     end    = digits + iint->_len;
-    while (digits < end) {
+
+    // iint's magnitude is stored in an array of 31-bit II_DIGIT values. Magnitudes that
+    // require 63 or 64 bits to represent will then take 3 II_DIGITS, only using 1 to 2
+    // bits from the most significant II_DIGIT.
+    const uint32_t max_digits_per_int64_t = 3;
+    const uint32_t whole_digits_per_int64_t = 2;
+    const uint32_t max_partial_digit_value = 3; // == 0b11, two populated bits
```

Comment:
These constants went through a few iterations before I settled on this.

I tried re-calculating these values based on `sizeof(int64_t)` and `II_BITS_PER_II_DIGIT`, but it was pretty verbose:

```c
const uint32_t bits_per_int64_t = sizeof(int64_t) * II_BITS_PER_BYTE;
const float digits_per_int64_t = (float) bits_per_int64_t / (float) II_BITS_PER_II_DIGIT;
const uint32_t max_digits = ceilf(digits_per_int64_t);
const uint32_t whole_digits = floorf(digits_per_int64_t);
const uint32_t max_bits_in_partial_digit = bits_per_int64_t % II_BITS_PER_II_DIGIT;
const uint32_t max_partial_digit_value = (1<<max_bits_in_partial_digit) - 1;
```

While a compiler would've probably been able to reduce these to zero-cost constants at build time, the complexity it introduced for folks trying to read the code felt too expensive.

Given that we're using a type name that will always be 64-bits (`int64_t` instead of, say, `long`), this would only ever need to be reevaluated if the definition of `II_BITS_PER_II_DIGIT` (currently `31`) ever changed, which won't happen any time soon. The code above didn't seem to carry its own weight. I opted to just use the resulting values instead.

I also tried defining macros for these values in `ion_int.h`, but it felt like overkill; readers would have to track down their definitions in another file despite this being the only place they're ever used.